### ISSUE TITLE
Proposal: Strongly Typed Cache Items

### ISSUE THREAD

**Issue by cocowalla** [Nodonisko/ionic-cache#52]
In PR #51 I introduced a method that allows users, with a single call, to get an existing cache item or generate and store a new one.

I also made that method strongly typed, such that it returns `T` and the value factory must produces a `T`.

I'd like to propose that _all_ get/set methods use typed values and parameters instead of `any`, so we can make better use of Typescript's type system.

**Comment by Nodonisko** [comment]
Thank you! I am on vacation I will check it on Monday :-)

Dne 18. 8. 2017 10:56 dop. napsal uživatel "cocowalla" <
notifications@github.com>:

In PR #51 <https://github.com/Nodonisko/ionic-cache/pull/51> I introduced a
method that allows users, with a single call, to get an existing cache item
or generate and store a new one.

I also made that method strongly typed, such that it returns T and the
value factory must produces a T.

I'd like to propose that *all* get/set methods use typed values and
parameters instead of any, so we can make better use of Typescript's type
system.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Nodonisko/ionic-cache/issues/52>, or mute the thread
